.github/workflows/ci.yml

```diff
@@ -82,6 +82,25 @@ jobs:
 
       - run: cargo package
 
+  msrv:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          # MSRV: Rust 1.47 and later have bugs in rustc that prevent some
+          # projects from upgrading past 1.46 yet. So, maintain compatibility
+          # for 1.46 until those bugs are fixed.
+          # Make sure to update rust-version in Cargo.toml when updating MSRV.
+          toolchain: 1.46.0
+
+      - run: |
+          cargo check --lib --all-features
+          cargo check --lib --no-default-features
```

Comment:
How could `--lib --no-default-features` fail to compile if `--lib --all-features` succeeded?

Reply:
I guess warnings from unused code only referenced from something feature-gated?

Reply:
But that's not MSRV-specific, right? We should catch that in the other jobs.

Reply:
Fair point; dropped that line